quodlibet/ext/events/musicbrainzsync.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2019 LoveIsGrief
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+from quodlibet import plugins, qltk
+from quodlibet.qltk.entry import UndoEntry
+from quodlibet.util.thread import call_async_background, Cancellable
+
+try:
+    import musicbrainzngs
+except ImportError:
+    raise plugins.MissingModulePluginException("musicbrainzngs")
+
+from gi.repository import Gtk
+from quodlibet import _
+from quodlibet.plugins import PluginConfig
+from quodlibet.plugins.events import EventPlugin
+
+ATTR_BRAINZ = 'musicbrainz_trackid'
+ATTR_RATING = '~#rating'
+
+BRAINZ_APP = "quodlibetMusicBrainzSync"
+VERSION = "0.1"
+
+plugin_config = PluginConfig("musicbrainz-sync")
+defaults = plugin_config.defaults
+defaults.set("username", "")
+defaults.set("password", "")
+
+
+class MusicBrainzSyncPlugin(EventPlugin):
+    PLUGIN_ID = "musicbrainzsync"
+    VERSION = VERSION
+    PLUGIN_NAME = _("MusicBrainz Sync")
+    PLUGIN_DESC = _("Syncs the rating of a song with music brainz.")
+
+    def __init__(self):
+        super(MusicBrainzSyncPlugin, self).__init__()
+        musicbrainzngs.set_rate_limit()
+        musicbrainzngs.set_useragent(
+            BRAINZ_APP, VERSION, "loveisgrief@tuta.io"
+        )
+        musicbrainzngs.auth(
+            plugin_config.get("username"),
+            plugin_config.get("password")
+        )
+
+    def plugin_on_changed(self, songs):
+        ratings_dict = {
+            song[ATTR_BRAINZ]: int(song[ATTR_RATING] * 100)
```

Comment:
Hmm, I didn't think `[]` for internal tags worked actually, but clearly I've forgotten. Nonetheless normally we use call semantics i.e. `song(ATTR_RATING)` as more processing sometimes happens, plus nobody likes `KeyError` :) )

Reply:
`KeyError` shouldn't happen since it's in a loop that checks if the attribute is in the song. I applied your suggestion anyway :slightly_smiling_face: